Dockerfile

```diff
@@ -16,24 +16,30 @@ RUN apt-get update && \
     libblas-dev libblocksruntime-dev libboost-dev libboost-filesystem-dev libboost-iostreams-dev \
     libboost-regex-dev libboost-system-dev libboost-thread-dev libbsd-dev libcairo2-dev \
     libcurl4-openssl-dev libcxsparse3 libedit-dev libffi6 libgmp10 libgnutls30 \
-    libhogweed4 libicu-dev libicu57 libidn11 libidn2-0 liblapack-dev libncurses5-dev libnettle6 \
+    libhogweed4 libicu-dev libicu63 libidn11 libidn2-0 liblapack-dev libncurses5-dev libnettle6 \
     libnss-myhostname libp11-kit0 libprotobuf-dev libpsl5 libsqlite3-dev libssl-dev libsuitesparse-dev \
-    libsystemd0 libtasn1-6 libudev1 libunistring0 libxml2-dev locales locate lsb-release lsof make \
+    libsystemd0 libtasn1-6 libudev1 libunistring2 libxml2-dev locales locate lsb-release lsof make \
     man-db mg mosh mutt net-tools netcat ninja-build pkg-config protobuf-compiler pstack rsync rsyslog \
     shunit2 socat ssh swift swig sysstat systemd-sysv systemtap-sdt-dev tcpdump tinc tmux traceroute \
     tzdata unattended-upgrades uuid-dev uuid-runtime wget whiptail xfsprogs libeigen3-dev && \
     apt-get clean autoclean && \
     apt-get autoremove -y && \
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
-
-#copy all the code related to C++ : use .dockerignore for unwanted files
-COPY ./src /anemomind/src

```

Comment:
We do require the src folder to build it

Reply:
I do not copy the sources here, but only after, on line 42.
That's because this way, if the sources change, docker will not recompile dependencies (poco, gtest, ceres), since it will know sources can influence dependencies.
Only if a relevant CMakeLists.txt is changed, then docker will recompile deps.